service/src/main/java/bio/terra/drshub/services/DrsProviderService.java

```diff
@@ -61,15 +61,17 @@ public record DrsProviderService(DrsHubConfig drsHubConfig) {
   public UriComponents getUriComponents(String drsUri) {
     UriComponents uriComponents;
 
-    var compactIdMatch = compactIdRegex.matcher(drsUri);
-    var hostNameMatch = hostNameRegex.matcher(drsUri);
+    // explicitly lowercase the uri because the case-insensitive regex flag is not working correctly
+    var drsUriLowerCase = drsUri.toLowerCase();
```

Comment:
Will this result in the DRS ID also being lowercased?
If so, that could potentially be an issue.
The [DRS (1.2+) specification](https://ga4gh.github.io/data-repository-service-schemas/preview/release/drs-1.2.0/docs/#section/DRS-IDs) allows uppercase characters in DRS IDs, and it whether the DRS server treats the DRS IDs as case sensitive or insensitive is an implementation detail left up to the DRS server.
Given this, it would be best to not case-shift the DRS ID.

As an alternate implementation, could just the compact identifier be lowercased in `getCompactIdUriComponents`?

Reply:
Yes it will, I was case-shifting the whole URI string to cover all my bases, but with the context that the IDs can be case-sensitive it make more sense to specifically shift the CIB only - I will update accordingly!